Sources/TokamakCore/CGStubs.swift

```diff
@@ -15,4 +15,193 @@
+///
+///     a   b   0
+///     c   d   0
+///     tx  ty  1
+public struct CGAffineTransform: Equatable {
+  public var a: CGFloat
+  public var b: CGFloat
+  public var c: CGFloat
+  public var d: CGFloat
+  public var tx: CGFloat
+  public var ty: CGFloat
+
+  /// The identity matrix
+  public static let identity: Self = .init(a: 1, b: 0, // 0
+                                           c: 0, d: 1, // 0
+                                           tx: 0, ty: 0) // 1
+
+  public init(a: CGFloat, b: CGFloat,
+              c: CGFloat, d: CGFloat,
+              tx: CGFloat, ty: CGFloat) {
+    self.a = a
+    self.b = b
+    self.c = c
+    self.d = d
+    self.tx = tx
+    self.ty = ty
+  }
+
+  /// Returns an affine transformation matrix constructed from a rotation value you provide.
+  /// - Parameters:
+  ///   - angle: The angle, in radians, by which this matrix rotates the coordinate system axes.
+  ///            A positive value specifies clockwise rotation and a negative value specifies counterclockwise rotation.
+  public init(rotationAngle angle: CGFloat) {
+    self.init(a: cos(angle), b: sin(angle), c: -sin(angle), d: cos(angle), tx: 0, ty: 0)
+  }
+
+  /// Returns an affine transformation matrix constructed from scaling values you provide.
+  /// - Parameters:
+  ///   - sx: The factor by which to scale the x-axis of the coordinate system.
+  ///   - sy: The factor by which to scale the y-axis of the coordinate system.
+  public init(scaleX sx: CGFloat,
+              y sy: CGFloat) {
+    self.init(a: sx, b: 0,
+              c: 0, d: sy,
+              tx: 0, ty: 0)
+  }
+
+  /// Returns an affine transformation matrix constructed from translation values you provide.
+  /// - Parameters:
+  ///   - tx: The value by which to move the x-axis of the coordinate system.
+  ///   - ty: The value by which to move the y-axis of the coordinate system.
+  public init(translationX tx: CGFloat,
+              y ty: CGFloat) {
+    self.init(a: 1, b: 0,
+              c: 0, d: 1,
+              tx: tx, ty: ty)
+  }
+
+  /// Transform the point into the transform's coordinate system.
+  public func transform(point: CGPoint) -> CGPoint {
+    CGPoint(x: (a * point.x) + (c * point.y) + tx,
+            y: (b * point.x) + (d * point.y) + ty)
+  }
+
+  /// Returns an affine transformation matrix constructed by combining two existing affine transforms.
+  /// - Parameters:
+  ///   - t2: The affine transform to concatenate to this affine transform.
+  /// - Returns: A new affine transformation matrix. That is, `t’ = t1*t2`.
+  public func concatenating(_ t2: Self) -> Self {
+    let t1m = [[a, b, 0],
+               [c, d, 0],
+               [tx, ty, 1]]
+    let t2m = [[t2.a, t2.b, 0],
+               [t2.c, t2.d, 0],
+               [t2.tx, t2.ty, 1]]
+    var res = [[CGFloat]]()
+    for i in 0..<3 {
+      for j in 0..<3 {
+        res[i][j] = 0
+        for k in 0..<3 {
+          res[i][j] += t1m[i][k] * t2m[k][j]
+        }
+      }
+    }
```

Comment:
I think this might be a hot path with performance degrading because of array allocations. I don't know of optimizing it would be premature though, especially in absence of any benchmarks.